admin/class-yoast-notification-center.php

```diff
@@ -582,7 +594,13 @@ public function update_storage() {
 		 *
 		 * @api Yoast_Notification[] $notifications
 		 */
-		$merged_notifications = apply_filters( 'yoast_notifications_before_storage', $merged_notifications );
+		$filtered_merged_notifications = apply_filters( 'yoast_notifications_before_storage', $merged_notifications );
+
+		// The notifications were filtered and therefore need to be stored.
+		if ( $merged_notifications !== $filtered_merged_notifications ) {
```

Comment:
I was not sure whether this would work as expected but it does.

This works because:
- When an array is assigned to a different variable it's values are actually copied, and not just the memory address of the array.
- The `!==` operator actually compares the values in the array instead of a memory address.